File: rendering/render_config.cpp
#include <string>
#include <iostream>
#include <cmath>

#include <directx/d3d12.h>
#include <directx/dxgi1_6.h>
#include <bullet/btBulletDynamicsCommon.h>

static const std::string RENDER_ENGINE  = "DirectX";
static const std::string PHYSICS_ENGINE = "BulletPhysics";
static const std::string SHADING_TYPE   = "PBR";
static const std::string TEXTURE_TYPE   = "PBR";

struct UVMap { float u, v; };
struct XYMap { float x, y; };
struct XZMap { float x, z; };

UVMap map_uv(float u, float v) { return { u, v }; }
XYMap map_xy(float x, float y) { return { x, y }; }
XZMap map_xz(float x, float z) { return { x, z }; }

void print_config() {
    std::cout << "Render Engine  : " << RENDER_ENGINE  << "\n";
    std::cout << "Physics Engine : " << PHYSICS_ENGINE << "\n";
    std::cout << "Shading Type   : " << SHADING_TYPE   << "\n";
    std::cout << "Texture Type   : " << TEXTURE_TYPE   << "\n";
}

File: rendering/render_config.ts
const RENDER_ENGINE: string  = "DirectX";
const PHYSICS_ENGINE: string = "BulletPhysics";
const SHADING_TYPE: string   = "PBR";
const TEXTURE_TYPE: string   = "PBR";

interface UVMap { u: number; v: number; }
interface XYMap { x: number; y: number; }
interface XZMap { x: number; z: number; }

const mapUV = (u: number, v: number): UVMap => ({ u, v });
const mapXY = (x: number, y: number): XYMap => ({ x, y });
const mapXZ = (x: number, z: number): XZMap => ({ x, z });

function printConfig(): void {
    console.log(`Render Engine  : ${RENDER_ENGINE}`);
    console.log(`Physics Engine : ${PHYSICS_ENGINE}`);
    console.log(`Shading Type   : ${SHADING_TYPE}`);
    console.log(`Texture Type   : ${TEXTURE_TYPE}`);
}

export { RENDER_ENGINE, PHYSICS_ENGINE, SHADING_TYPE, TEXTURE_TYPE, mapUV, mapXY, mapXZ, printConfig };
